h/api/views.py

```diff
@@ -201,20 +202,37 @@ def read_jsonld(annotation, request):
     return presenter.asdict()
 
 
-@api_config(route_name='api.annotation', request_method='PUT', permission='update')
+@api_config(route_name='api.annotation',
+            request_method='PUT',
+            permission='update')
 def update(annotation, request):
     """Update the specified annotation with data from the PUT payload."""
-    schema = schemas.LegacyUpdateAnnotationSchema(request,
-                                                  annotation=annotation)
-    appstruct = schema.validate(_json_payload(request))
-    annotation = storage.legacy_update_annotation(request,
-                                                  annotation.id,
-                                                  appstruct)
+    json_payload = _json_payload(request)
 
-    _publish_annotation_event(request, annotation, 'update')
+    legacy_annotation = elastic.Annotation.fetch(annotation.id)
 
-    presenter = AnnotationJSONPresenter(request, annotation)
-    return presenter.asdict()
+    # Validate the annotation for, and update the annotation in, PostgreSQL.
+    if request.feature('postgres'):
+        schema = schemas.UpdateAnnotationSchema(request, annotation=annotation)
+        appstruct = schema.validate(copy.deepcopy(json_payload))
+        annotation = storage.update_annotation(request.db,
+                                               annotation.id,
+                                               appstruct)
```

Comment:
You were saying that we support partial updating of annotations. We might have to build something that makes sure that partial updating doesn't fail in case we don't have the annotation in postgres yet. Although we probably will only miss a very small amount of the newest annotations for a short period after switching on the feature and until the last migration run has finished.